### ISSUE TITLE
Add ability to customize shortcut key

### ISSUE THREAD

**Issue by hackel** [reblws/tab-search#2]
I would like the ability to define a custom shortcut key to activate the extension.  Perhaps this is something Firefox needs to implement, the same way that Chrome has a page to manage hotkeys for extensions.

**Comment by reblws** [comment]
I don't think this is possible right now, it doesn't look like Firefox has any support for changing extension hotkeys the same way Chrome does. Is there anything particular about the current hotkey that doesn't work for you?

**Comment by blakebutcher** [comment]
I'd also like this. My preference would be CMD+E to make it consistent with Vivaldi's Quick Commands feature.

**Comment by hackel** [comment]
In terms of why I'd like to change it, I would just like it to be consistent with other editors I use.  Also, I just noticed that, at least on GitHub while I have this input focused, ctrl-shift-L inserts the following:
```

- [ ] 
```

**Comment by reblws** [comment]
@hackel 
Okay, that's pretty bad, I'll change it since I don't want the extension to interfere with anyone's workflow. 

I originally chose Ctrl+Shift+L because it was closest to Ctrl+Shift+P (the Sublime Text default, currently opens incognito tabs in FF) and the idea was that those three keys would get your hands ready in a typing position.

I'm most likely going to switch the shortcut to Ctrl+E in the next couple of days, unless there are any objections or problems I find with it.

**Comment by ScriptSmith** [comment]
I quite like `Ctrl`+`Shift`+`A` which is the Intellij / Jetbrains* equivalent, however it seems that combination either selects all text, or opens Firefox's addons page depending on context.

I also read through the Mozilla's WebExtension docs and it does seem that the only way to customize shortcuts is to include duplicate entries in the manifest and use multiple command listeners - not really an ideal solution. 

I'll try out `Ctrl` +`E` when the change is made, and if It doesn't work out I'll probably try out patches with other combinations.

**Comment by reblws** [comment]
After trying Ctrl+E out, I'm concerned that it's way too close to the destructive commands (Ctrl+W - delete current tab, Ctrl+Q close entire window), the cost of missing the shortcut would be too much -- a tired user might accidentally delete their current browser window or close their current tab when all they wanted to do was navigate their current tab. Also current users might be confused about the old shortcut suddenly changing.

Maybe it wouldn't be so bad to have multiple command listeners the way @ScriptSmith mentioned -- I'd define Ctrl+Shift+L available as the default browser action and Ctrl+E as an alternate keybinding that's left on by default but can be turned off in the settings, at least until Mozilla gets around to implementing custom shortcuts.

Here are the relevant bugzillas to watch out for:

https://bugzilla.mozilla.org/show_bug.cgi?id=1320332
https://bugzilla.mozilla.org/show_bug.cgi?id=1303384

**Comment by reblws** [comment]
Ok after fumbling around with the shortcuts, it looks like (browserAction.openPopup)[https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/browserAction/openPopup] doesn't actually work if it's under a command listener. I'll be switching the shortcut to Ctrl+E in the meantime and add a shortcut hint to search placeholder text.

**Comment by thomcc** [comment]
In nightly it should be possible to implement this, now that [bug 1421811](https://bugzilla.mozilla.org/show_bug.cgi?id=1421811) is fixed. There's even a [handy library](https://github.com/piroor/webextensions-lib-shortcut-customize-ui) for it (which at the very least can serve as example code).

**Comment by reblws** [comment]
Thanks @thomcc I'm looking into it.

**Comment by nikitavoloboev** [comment]
So what is the shortcut to trigger Tab Search on macOS? I can't find it anywhere in the settings.

**Comment by cwagner22** [comment]
Hey @nikitavoloboev, it's ⌘⇧L but I'm using Karabiner-Elements to remap it for now

**Comment by nikitavoloboev** [comment]
Thanks @cwagner22. I tried ⌘⇧L but realized that I had to have the icon in the toolbar for it to work. Now it works. 🙂

**Comment by reblws** [comment]
Hi all, sorry I haven't had the time to update the extension with this helpful feature. It'll be in the next update when I get around to it.

**Comment by chodorowicz** [comment]
cmd + shift + L doesn't work for some reason on SoundCloud tabs

**Comment by HaySayCheese** [comment]
Guys, any plans to provide ability to change `Ctrl+Shift+F` to some other shortcut? Unf. `Ctrl+Shift+F` collides with Google Docs (channging layout of the interface).

**Comment by reblws** [comment]
Hi Dima, sorry I'm not actively maintaining this extension at the moment.
What you need right now is possible in the latest versions of Firefox (I
believe it's in the latest ESR release but I'm not 100% sure).  I just
haven't added a convenient component to modify it from the regular settings
page.

In order to change the shortcut:

Go to Add-ons Manager > Hit the Gear button > Click "Manage Shortcut
Extensions"

You should be able to see a list of all the extensions you have installed
and their main shortcuts. You can change the default shortcut from there.

Enrico

**Comment by pjbuterbaugh** [comment]
Ctrl + Shift + F collides with "View in Full Screen/Immersive Editor" for GSuite.
